cockpit/src/subscriptions-client.js

```diff
@@ -121,9 +122,27 @@ function getSubscriptionDetails() {
     });
 }
 
+/* convenience function for specifying d-bus strings */
+function s(value) {
+    return {
+        t: 's',
+        v: value,
+    };
+}
+
 client.registerSystem = subscriptionDetails => {
     const dfd = cockpit.defer();
-    const connection_options = {};
+    // Note: when values are not specified we force use of default
+    // values. Otherwise old and obsolete values from rhsm.conf could be used.
+    const connection_options = {
+        host: s(RHSM_DEFAULTS.hostname),
+        port: s(RHSM_DEFAULTS.port),
+        proxy_hostname: s(''),
+        proxy_port: s(''),
+        proxy_user: s(''),
+        proxy_password: s(''),
+        handler: s(RHSM_DEFAULTS.prefix),
+    };
```

Comment:
It simplifies code, but it breaks resetting of config values. The `if` statement on the line 299 is always true.

Reply:
> It simplifies code, but it breaks resetting of config values. The if statement on the line 299 is always true.

Hmmm. I'm thinking this is a not a real issue in terms of functionality, but I'll make a commit to remove dead code/simplify.

I'm leaving explanation in the change.